code/modules/projectiles/ammunition.dm

```diff
@@ -139,6 +138,9 @@
 		stored_ammo -= b
 		if(keep)
 			stored_ammo.Insert(1,b)
+		if(!mats_initialized)
+			initial_mats = materials.Copy()
+			mats_initialized = TRUE
```

Comment:
Took me a bit to understand why you added this logic here and not somewhere else.
The autolathe (and other lathes) change the material value after the object is `new`ed.

I think it would be better to put this in `update_mat_value` at the start of the proc. Then you are 100% sure all cases will go correct as long as the code calls `update_mat_value` when the bullet count updates.
Could even remove the extra variable `mats_initialized` by checking if `initial_mats` is null instead.

Reply:
Good call on the mats initialized, but the update_mat_value has to go after, since that uses the initial_mats to calculate.

Reply:
I meant that the initialize has to happen in the `update_mat_value` proc itself. The first thing it does in fact.
This way the order of execution stays the same (init first if needed, then update) and you cover all possible paths properly